django/contrib/admin/options.py

```diff
@@ -143,7 +143,7 @@ def formfield_for_dbfield(self, db_field, request, **kwargs):
                 kwargs = dict(self.formfield_overrides[db_field.__class__], **kwargs)
 
             # Get the correct formfield.
-            if isinstance(db_field, models.ForeignKey):
+            if db_field.many_to_one:
                 formfield = self.formfield_for_foreignkey(db_field, request, **kwargs)
```

Comment:
Here if a field is not a ForeignKey, but have many_to_one=True then it will go through formfield_for_foreignkey().

If that method returns None, then [None will be returned](https://github.com/django/django/pull/6236/files#diff-3c42de3e53aba87b32c494f995a728dfR168) and [db_field.formfield()](https://github.com/django/django/pull/6236/files#diff-3c42de3e53aba87b32c494f995a728dfR178) won't have a chance to be called.

In this case, the formfield [will be ignored by ModelForm](https://github.com/django/django/blob/master/django/forms/models.py#L182-L185).

Shouldn't we have a way to use the custom field's formfield() method even if it has many_to_one=True ?

Note that this concerned is similar to the one raised in #24227